sormas-ui/src/main/java/de/symeda/sormas/ui/samples/PathogenTestForm.java

```diff
@@ -222,13 +222,20 @@ protected void addFields() {
 			Arrays.asList(PathogenTestType.CQ_VALUE_DETECTION),
 			true);
 
-		diseaseField.addValueChangeListener((ValueChangeListener) valueChangeEvent -> {
-			Disease disease = (Disease) valueChangeEvent.getProperty().getValue();
+		Consumer<Disease> updateDiseaseVariantField = disease -> {
```

Comment:
Is there a specific reason to have a `Consumer` here instead of a method `PathogenTestForm.updateDiseaseVariantField`?

Reply:
I preferred a Consumer because

- the code fragment is short (only three statements)
- it's just used in one method
- with a consumer, the logic is all in one place